server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/ProjectAssociationActions.java

```diff
@@ -0,0 +1,147 @@
+          profileProjectOperations.addProject(profileKey, projectUuid, userSession);
+        }
+      });
+    setCommonAttributes(addProject);
+
+    NewAction removeProject = controller.createAction("remove_project")
+      .setSince("5.2")
+      .setDescription("Remove a project's association with a quality profile")
+      .setHandler(new AssociationHandler(profileLookup, componentService) {
+        @Override
+        protected void changeAssociation(String profileKey, String projectUuid, UserSession userSession) {
+          profileProjectOperations.removeProject(profileKey, projectUuid, userSession);
+        }
+      });
+    setCommonAttributes(removeProject);
+  }
+
+  private void setCommonAttributes(NewAction action) {
+    action.setPost(true);
+    action.createParam(PARAM_PROJECT_UUID)
+      .setDescription("A project UUID. Either this parameter, or projectKey must be set");
+    action.createParam(PARAM_PROJECT_KEY)
+      .setDescription("A project key. Either this parameter, or projectUuid must be set");
+    action.createParam(PARAM_PROFILE_KEY)
+      .setDescription("A quality profile key. Either this parameter, or a combination of profileName + language must be set");
+    action.createParam(PARAM_PROFILE_NAME)
+      .setDescription("A quality profile name. If this parameter is set, profileKey must not be set and language must be set to disambiguate");
+    action.createParam(PARAM_LANGUAGE)
+      .setDescription("A quality profile language. If this parameter is set, profileKey must not be set and language must be set to disambiguate")
+      .setPossibleValues(Collections2.transform(Arrays.asList(languages.all()), new Function<Language, String>() {
+        @Override
+        public String apply(Language input) {
+          return input.getKey();
+        }
+      }));
+  }
+
+  private static abstract class AssociationHandler implements RequestHandler {
+
+    private final QProfileLookup profileLookup;
+    private final ComponentService componentService;
+
+    public AssociationHandler(QProfileLookup profileLookup, ComponentService componentService) {
+      this.profileLookup = profileLookup;
+      this.componentService = componentService;
+    }
+
+    protected abstract void changeAssociation(String profileKey, String projectUuid, UserSession userSession);
+
+    @Override
+    public void handle(Request request, Response response) throws Exception {
+      String language = request.param(PARAM_LANGUAGE);
+      String profileName = request.param(PARAM_PROFILE_NAME);
+      String profileKey = request.param(PARAM_PROFILE_KEY);
+      String projectKey = request.param(PARAM_PROJECT_KEY);
+      String projectUuid = request.param(PARAM_PROJECT_UUID);
+
+      Preconditions.checkArgument(
+        (language != null && profileName != null) ^ profileKey != null, "Either profileKey or profileName + language must be set");
```

Comment:
What happens if values are empty but not null ? For example profileKey=&projectKey=